ygen/protogen_test.go

```diff
@@ -654,6 +654,102 @@ func TestGenProto3Msg(t *testing.T) {
 				}},
 			},
 		},
+	}, {
+		name: "simple message with only scalar fields and field number extensions",
+		inMsg: &Directory{
+			Name: "MessageName",
+			Entry: &yang.Entry{
+				Name: "message-name",
+				Dir:  map[string]*yang.Entry{},
+				Kind: yang.DirectoryEntry,
+			},
+			Fields: map[string]*yang.Entry{
+				"field-one": {
+					Name: "field-one",
+					Type: &yang.YangType{Kind: yang.Ystring},
+					Exts: []*yang.Statement{
+						{
+							Keyword:  "occodegenext:field-number-offset",
+							Argument: "100",
+						},
+					},
+				},
+				"field-two": {
+					Name: "field-two",
+					Type: &yang.YangType{Kind: yang.Yint8},
+					Exts: []*yang.Statement{
+						{
+							Keyword:  "occodegenext:field-number",
+							Argument: "1",
+						},
+					},
+				},
+				"field-three": {
+					Name: "field-three",
+					Type: &yang.YangType{Kind: yang.Yint8},
+					Exts: []*yang.Statement{
+						{
+							Keyword:  "occodegenext:field-number",
+							Argument: "1",
+						},
+						{
+							Keyword:  "occodegenext:field-number-offset",
+							Argument: "100",
+						},
+					},
+				},
+			},
+			Path: []string{"", "root", "message-name"},
+		},
+		inBasePackage: "base",
+		inEnumPackage: "enums",
+		wantMsgs: map[string]*protoMsg{
+			"MessageName": {
+				Name:     "MessageName",
+				YANGPath: "/root/message-name",
+				Fields: []*protoMsgField{{
+					Tag:  410095931,
+					Name: "field_one",
+					Type: "ywrapper.StringValue",
+				}, {
+					Tag:  1,
+					Name: "field_two",
+					Type: "ywrapper.IntValue",
+				}, {
+					Tag:  101,
+					Name: "field_three",
+					Type: "ywrapper.IntValue",
+				}},
+			},
+		},
+	}, {
+		name: "message with field number in reserved range",
```

Comment:
In my reading of the spec only the generated values are required to not fall within 1-1000, but practically speaking manually-set values should be restricted to 1-1000 to avoid potential collisions. Makes sense to me to restrict manually set values to fall within 1-1000.